### PR TITLE
feat: calculate branch reusable workflow

### PR DESCRIPTION
**Pull request by paulredmond**
## Description

Reusable workflow to validate if a given branch exists in a repo. If the repo contains the branch, the action sends the sha and a short version of the sha as output.

**Jira Issue:** https://revolutionparts.atlassian.net/browse/DEVX-655

## Background
When deploying QA environments, the operator will provide a branch to deploy. Each project in the deployment will build the given branch if it exists, or deploy the `latest` Dockerfile otherwise.

## Testing Information
I am already testing this through internal_api. Here's an action run using this workflow https://github.com/encodium/internal_api/actions/runs/6804578886

### CODE REVIEW

File: .github/workflows/calculate-branch.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            echo "Branch \"${branch_name_candidate}\" found."
+            branch_name="${branch_name_candidate}"
+            sha=$(echo "${branch_info}" | jq -r '.sha')
+            sha_short=$(echo "${branch_info}" | jq -r '.sha[:7]')
+            found=true
+          else
+            branch_name=""
+            sha=""
+            sha_short=""
+            found=""
```
Comment (aspencer) [inline .github/workflows/calculate-branch.yaml:69]: I wonder if we should fill this out more (found=false etc) - does this get additional information if the branch name is not found, or is it just empty?

**Review by aspencer — APPROVED** [review]

File: .github/workflows/calculate-branch.yaml (hunk 1 above)
Comment (paulredmond) [reply .github/workflows/calculate-branch.yaml:69]: I think empty for the first three is good, but I like the idea of doing `found=false` or `found=0|1`. Bash doesn't really have a concept of boolean so it's up to the user to validate `== 'true'` or `== 1`, for example.

Comment (paulredmond) [reply .github/workflows/calculate-branch.yaml]: Updated to `found=false`, thanks!

Comment (paulredmond) [reply .github/workflows/calculate-branch.yaml]: When it doesn't find a branch, here's the output:
image

**Review by aspencer — APPROVED** [review]

**Pull request merged**